index.js

```diff
@@ -1270,78 +947,15 @@
                    rejectInconsistent);
     };
 
-    //  paramTypesAndValues :: ((a -> Any), Type, [a]) -> ErrorRepr
-    var paramTypesAndValues = function(valFn, type, values) {
-      return map(
-        chain(values, valFn),
-        function(value) {
-          return typesAndValues(
-            type,
-            [value]
-          );
-        }
-      );
-    };
-
-    //  nullaryTypesAndValues :: [Any] -> ErrorRepr
-    var nullaryTypesAndValues = function(values) {
-      return {
-        isParameterizedType: false,
-        types: valueTypes(values),
-        values: values
-      };
-    };
-
-    //  typesAndValues :: (type, [Any]) -> ErrorRepr
-    var typesAndValues = function(t, values) {
-      if (isParameterizedType(t)) {
-        var constructorName = functionName(values[0].constructor);
```

Comment:
This has a few shortcomings:

```javascript
//    Identity :: a -> Identity a
const Identity = x => ({'@@type': 'my-package/Identity', value: x});

Identity(42).constructor.name;
// => 'Object'

S.Just(42).constructor.name;
// => 'Maybe'
```

We'd like to use `Identity( … )` and `Just( … )` in these cases to provide context, but I don't see how we can do so.